### ISSUE TITLE
[Inline Editable] Add a label to the component's edit button (CAL-AMP29)

### ISSUE THREAD

**Issue by dqateam** [Esri/calcite-design-system#13534]
**monday.com sync:** #10821967254

### Violation: 
 Ensure headings and labels are descriptive and unique 

<img width="1719" height="228" alt="Image" src="https://github.com/user-attachments/assets/1f3ba30a-8952-4f37-91cb-acb8654e1f92" />

<img width="1719" height="228" alt="Image" src="https://github.com/user-attachments/assets/e726ebf8-7478-4778-b995-377eb3fcb3d5" />

### Recommended solution
Add a label, that can be translated, to `inline-editable`'s edit button, similar to how the `title` is populated.

### WCAG Reference: 
- [2.4.6 Headings and Labels](https://www.w3.org/WAI/WCAG22/Understanding/headings-and-labels.html) 

### Severity: 
 Low 

### Media Type: 
 Page Structure 

--- 
### Areas for Remediation: 
- Title: CAL-AMP29 Ensure headings and labels are descriptive and unique 
- Module: M4 Form Elements_Employee Dashboard 
- Finding ID: [CAL-AMP29](https://esri.hub.essentia11y.com/short-link/_O02bBqhW6ENHKwX) 

**Issue**
There are buttons with insufficiently descriptive accessible names. Examples include:
- "Edit" (Checkmark present beside "zip code" section)
- "?" (present beside "Regional office").

**User Impact**
Screen reader users will be unable to determine the purpose of these buttons.

**Code Reference** 
 ```
<button aria-live="polite" aria-busy="false" aria-label="Edit" class=" button-padding--shrunk icon-end-empty " href="false" rel="false" target="false" type="button">(...)</button>

```

**Steps to Reproduce**
1. Locate the button.
2. Inspect it with Chrome DevTools.
3. In the Accessibility tab, expand the Computed Properties section.
4. Review the value for "Name".
 
**Recommendation**
Ensure headings and labels are descriptive and unique. This includes the accessible names of buttons. The accessible name of a button can be set with internal text, an aria-label attribute, or an aria-labelledby attribute. Good accessible names are both concise and descriptive. Avoid including the word "button" in the accessible name, as this information is already supplied by the button's role.

**Compliant Code Example**
```
<!-- example of headings that describe topic or purpose -->
<h1>Dashboard</h1> 
<h2>Reporting</h2>
<h3>Reporting Details </h3>
<h2>Testing</h2> 
<h3>Testing Details</h3>
```

 **Issue Image** 
 https://desktopqa/aic_lap/Calcite/Calcite2025/CAL-AMP29.png 
 

 --- 
### Additional Resources: 
- [LA Platform Report Link](https://esri.hub.essentia11y.com/short-link/_O02bBqhW6ENHKwX) 
- [Esri Accessibility Resources](https://esriis.sharepoint.com/sites/Accessibility)
- Report Source: Product accessibility evaluation conducted on Calcite by Level Access and delivered August, 2025.

**Comment by ashetland** [comment]
@Esri/calcite-pes Should this be incorporated into #6153?

**Comment by geospatialem** [comment]
> [@Esri/calcite-pes](https://github.com/orgs/Esri/teams/calcite-pes) Should this be incorporated into [#6153](https://github.com/Esri/calcite-design-system/issues/6153)?

For a11y tracking we should try to keep the above open, @ashetland could we add it as acceptance criteria for #6153?

**Comment by ashetland** [comment]
Makes sense to me!

**Comment by DintaMel** [comment]
@Esri/calcite-pes When I inspect the Accessibility tab and when Jaws reads off the edit button name, it is "edit edit". I checked Inline Editable's Save and Cancel buttons and saw similar repetition. I also saw the same thing in Action, but not on Button. Should the expectation be that edit is repeated? See examples below.

[Inline Editable](https://next.gha.afd.arcgis.com/calcite-design-system/components/inline-editable/#sample) - edit button
<img width="634" height="665" alt="Image" src="https://github.com/user-attachments/assets/3740f186-214f-4577-8c88-82e2dfd42a85" />

[Action](https://next.gha.afd.arcgis.com/calcite-design-system/components/action/#sample)

<img width="855" height="543" alt="Image" src="https://github.com/user-attachments/assets/07bf5e7c-b99b-49bb-b42a-29a0457d25a3" />


[Button](https://next.gha.afd.arcgis.com/calcite-design-system/components/button/#sample)

<img width="493" height="466" alt="Image" src="https://github.com/user-attachments/assets/49b1ab44-6bc9-4888-818e-f295758f7850" />

**Comment by geospatialem** [comment]
> [@Esri/calcite-pes](https://github.com/orgs/Esri/teams/calcite-pes) When I inspect the Accessibility tab and when Jaws reads off the edit button name, it is "edit edit". I checked Inline Editable's Save and Cancel buttons and saw similar repetition. I also saw the same thing in Action, but not on Button. Should the expectation be that edit is repeated? See examples below.
> 
> [Inline Editable](https://next.gha.afd.arcgis.com/calcite-design-system/components/inline-editable/#sample) - edit button <img alt="Image" width="634" height="665" src="https://private-user-images.githubusercontent.com/18604328/628778725-3740f186-214f-4577-8c88-82e2dfd42a85.png?jwt=eyJ0eXAiOiJKV1QiLCJhbGciOiJIUzI1NiJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3ODU0MjI4MDgsIm5iZiI6MTc4NTQyMjUwOCwicGF0aCI6Ii8xODYwNDMyOC82Mjg3Nzg3MjUtMzc0MGYxODYtMjE0Zi00NTc3LThjODgtODJlMmRmZDQyYTg1LnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNjA3MzAlMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjYwNzMwVDE0NDE0OFomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPTcxNDkxY2U3NjQ4MmNlZWQyZWFiNzcwY2Q0OWNlNDRlNGExYjAzYjNlOWUyZThmM2UzN2FhMjE1MGQ2MzNmNTcmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0JnJlc3BvbnNlLWNvbnRlbnQtdHlwZT1pbWFnZSUyRnBuZyJ9.ci_hY04om-9sssMrItCJu4itdMAHSzLAXu2D2G-VwLw">
> 
> [Action](https://next.gha.afd.arcgis.com/calcite-design-system/components/action/#sample)
> 
> <img alt="Image" width="855" height="543" src="https://private-user-images.githubusercontent.com/18604328/628780632-07bf5e7c-b99b-49bb-b42a-29a0457d25a3.png?jwt=eyJ0eXAiOiJKV1QiLCJhbGciOiJIUzI1NiJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3ODU0MjI4MDgsIm5iZiI6MTc4NTQyMjUwOCwicGF0aCI6Ii8xODYwNDMyOC82Mjg3ODA2MzItMDdiZjVlN2MtYjk5Yi00OWJiLWI0MmEtMjlhMDQ1N2QyNWEzLnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNjA3MzAlMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjYwNzMwVDE0NDE0OFomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPWY2Y2NjNDc2Y2U5YTZlZDczODgxNTZkNzMyNTFiODIzNjNiZjcwZGVkYjk0NzYzZTAzYjg3ZjgxNTBiNTc2OGEmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0JnJlc3BvbnNlLWNvbnRlbnQtdHlwZT1pbWFnZSUyRnBuZyJ9.vCy5HpN2KZDunF5lCm_4vpdZnA4eCqt3EHEv1LnoNDU">
> [Button](https://next.gha.afd.arcgis.com/calcite-design-system/components/button/#sample)
> 
> <img alt="Image" width="493" height="466" src="https://private-user-images.githubusercontent.com/18604328/628780872-49b1ab44-6bc9-4888-818e-f295758f7850.png?jwt=eyJ0eXAiOiJKV1QiLCJhbGciOiJIUzI1NiJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3ODU0MjI4MDgsIm5iZiI6MTc4NTQyMjUwOCwicGF0aCI6Ii8xODYwNDMyOC82Mjg3ODA4NzItNDliMWFiNDQtNmJjOS00ODg4LTgxOGUtZjI5NTc1OGY3ODUwLnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNjA3MzAlMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjYwNzMwVDE0NDE0OFomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPWFkY2I5Y2FjOGExOTVkZjgyYWIzYzNhZjk3NWY0NTIzNjA5NTYzM2NhYjY5NDdiMDQ3MDRmZmI1YzNlM2JmYzAmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0JnJlc3BvbnNlLWNvbnRlbnQtdHlwZT1pbWFnZSUyRnBuZyJ9.8Gv-ZI9zN60bfZF8i-DMqoKoBY3YjxHQLBzJUP3p1J4">

The above appears to be repeating `action`'s `label` and `text` attributes, which are both supplied via the update. This shouldn't be repeated as the label is supposed to generate the context to AT, and would be out of scope for the above PR. 

When testing with https://codepen.io/geospatialem/pen/WbRYzjz, it appears this occurred between `5.1` and `next`... and seems related to https://github.com/Esri/calcite-design-system/pull/14690, which resulted in the opening of https://github.com/Esri/calcite-design-system/issues/14825. I'll work on prioritizing the effort, but the above appears to be verified with the additions of the label. The follow-up regression should be addressed before `5.2`.

**Comment by geospatialem** [comment]
Marking the above as verified, with a follow-up for `action` with #14825.
